integration/src/main/java/com/aws/greengrass/mqtt/moquette/ClientDeviceAuthorizer.java

```diff
@@ -126,28 +125,30 @@ public boolean canWrite(Topic topic, String user, String client) {
     @Override
     public boolean canRead(Topic topic, String user, String client) {
         String resource = "mqtt:topicfilter:" + topic;
-        boolean canPerform = false;
-        try {
-            canPerform = canDevicePerform(getOrCreateSessionForClient(client, user), "mqtt:subscribe", resource);
-        } catch (AuthenticationException e) {
-            LOG.atWarn().cause(e).kv(CLIENT_ID, client).log("Unable to re-authenticate client.");
-        }
+        boolean canPerform = canDevicePerform(client, user, "mqtt:subscribe", resource);
         LOG.atDebug().kv("topic", topic).kv("isAllowed", canPerform).kv(CLIENT_ID, client)
             .log("MQTT subscribe request");
         return canPerform;
     }
 
-    private boolean canDevicePerform(UserSessionPair sessionPair, String operation, String resource) {
-        if (sessionPair == null) {
-            return false;
-        }
+    private boolean canDevicePerform(String client, String user, String operation, String resource) {
         boolean canPerform = false;
         try {
-            canPerform = canDevicePerform(sessionPair.getSession(), operation, resource);
-        } catch (InvalidSessionException e) {
-            LOG.atError().kv(SESSION_ID, sessionPair.getSession()).cause(e).log("Session ID is invalid");
-        } catch (AuthorizationException e) {
-            LOG.atError().kv(SESSION_ID, sessionPair.getSession()).cause(e).log("Authorization Exception");
+            UserSessionPair sessionPair = getOrCreateSessionForClient(client, user);
+            if (sessionPair == null) {
```

Comment:
I copied the null check from the old `canDevicePerform`, but it does seem like an unnecessary check since we are either getting a current session or creating a new one. I'll remove the null check since the sessionPair will never be null.